src/main/java/network/brightspots/rcv/Utils.java

```diff
@@ -86,4 +93,16 @@ static String getUserName() {
   static String[] splitByNewline(String s) {
     return s.trim().split("\\s*\\r?\\n\\s*");
   }
+
+  static String bytesToHex(byte[] hash) {
+    StringBuilder hexString = new StringBuilder(2 * hash.length);
+    for (byte b : hash) {
+      String hex = Integer.toHexString(0xff & b);
+      if (hex.length() == 1) {
+        hexString.append('0');
+      }
```

Comment:
could you explain this part?

Reply:
`toHexString` strips all leading zeros for some reason, so byte 0 would be hex `0` instead of `00`

Reply:
I'm probably just confused here, but why aren't we prepending '0' instead of appending it?

Reply:
We append `0` before we append the remainder of the string, so it's effectively prepending!

Reply:
Sigh. Reading comprehension!